src/components/TreeSelect/Dropdown.tsx

```diff
@@ -0,0 +1,90 @@
+import { useState } from "react";
+import styled from "@emotion/styled";
+import Icon from "@leafygreen-ui/icon";
+import { uiColors } from "@leafygreen-ui/palette";
+
+const { gray } = uiColors;
+
+interface RenderProps {
+  isVisible: boolean;
+  setOptionsLabel?: React.Dispatch<React.SetStateAction<string>>;
+}
+
+interface Props {
+  inputLabel: string;
+  "data-cy": string;
+  width?: string;
+  render: ({
+    getDropdownProps,
+  }: {
+    getDropdownProps: () => RenderProps;
+  }) => JSX.Element;
+}
+
+// including a TreeDataEntry with value = "all"
+// will serve as the 'All' button
+export const Dropdown: React.FC<Props> = ({
+  inputLabel, // label for the select
+  "data-cy": dataCy, // for testing only
+  width,
+  render,
+}) => {
+  const [isVisible, setisVisible] = useState(false);
+  const toggleOptions: () => void = () => setisVisible(!isVisible);
+  const [optionsLabel, setOptionsLabel] = useState(null);
+
+  const getDropdownProps = () => ({
+    isVisible,
+    setOptionsLabel,
+    isDropdown: true,
+  });
+
+  return (
+    <Wrapper data-cy={dataCy} width={width}>
+      <BarWrapper onClick={toggleOptions} className="cy-treeselect-bar">
```

Comment:
"cy-treeselect-bar" should belong to the data-cy attribute instead of className. I made a ticket so we can do it later on since it will require more test changes: https://jira.mongodb.org/browse/EVG-14933